### ISSUE TITLE
`$SecureString |Should -be` testable

### ISSUE THREAD

**Issue by iRon7** [pester/Pester#2374]
### Checklist

- [X] Feature request has a meaningful title
- [X] I have searched the existing issues. See [all issues](https://github.com/pester/Pester/issues?q=is%3Aissue)
- [X] I have tested using the latest version of Pester. See [Installation and update guide](https://pester.dev/docs/introduction/installation).

### Summary of the feature request

Not just from a "*ease of use*"  (which is my current use case), but also from a security perspective where I might not want to reveal the plaintext secret (to e.g. keep it away from possible logging etc.), I would like to be able to directly test a secure string (hash) without decrypting it:

```PowerShell
$Password = ConvertTo-SecureString 'P@ssW0rD!' -AsPlainText -Force
```

```PowerShell
$SecureString = ConvertTo-SecureString '01000000d08c9ddf0115d1118c7a00c04fc297eb01000000d63fe8e2130898439bc1614dcea3f2c70000000002000000000010660000000100002000000088771547cc1125987fff88a0b77ba5596a4133ad9b3c862d51a69d1762126ce2000000000e800000000200002000000012557d0d0a101fcfd27050e459ed19e9b21224578ee35c162eed57b75f8bde96200000005c1b01787b3b7a931c84314c5a81a5973b8341da4e715247a58970ef4b327cab400000003c1ea6534eb066590b9082a8eb16cd6b08fc9898bb0108065cdb822888edcd936925cdb96f8c3c539a1e0003fe0d40c6ff3a3765d328a8dc819c9681559b4db9'

$Password |Should -be $SecureString
```

### Expected
No response (`$Password` and `$SecureString` are qual)

### Actual
Error:
```
Expected System.Security.SecureString, but got System.Security.SecureString.
At C:\Program Files\WindowsPowerShell\Modules\Pester\5.4.1\Pester.psm1:8140 char:13
+             throw $errorRecord
+             ~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidResult: (System.Collecti...,System.Object]:Dictionary`2) [], Exception
    + FullyQualifiedErrorId : PesterAssertionFailed
```

**Comment by fflaten** [comment]
This isn't possible without decrypting. A secure string is a memory reference, so even a copy won't match.
```powershell
$s1 = ConvertTo-SecureString 'abc123' -AsPlainText -Force
$s2 = ConvertTo-SecureString 'abc123' -AsPlainText -Force

$s1 -eq $s1
# True

$s1 -eq $s2
# False

$s1copy = $s1.Copy()
$s1 -eq $s1copy
# False
```

You'd also need to have a shared secret to make this portable since `ConvertFrom-SecureString` by default encrypts using DPAPI which is associated to current user on current machine, at which point you have another secret to manage. 

This is better suited as a [custom assertion](https://pester.dev/docs/assertions/custom-assertions) imo.

**Comment by iRon7** [comment]
@fflaten 

> This isn't possible without decrypting. A secure string is a memory reference, so even a copy won't match

Agree, but (I think) it safer to this in the assertion rather than in the Pester test code.

> by default only encrypts on Windows using DPAPI

Agree, didn't think of that. I will close the issue.

> Complex assertions like this are better suited as a [custom assertion](https://pester.dev/docs/assertions/custom-assertions) imo.

Thanks for link (I didn't even know that was possible).
For what it is worth, see: https://github.com/iRon7/Should-BeSecureString
